src/meta/coinductive_predicates.lean

```diff
@@ -52,13 +51,51 @@ meta def add_theorem_by (n : name) (ls : list name) (type : expr) (tac : tactic
   add_decl $ mk_theorem n ls type body,
   return $ const n $ ls.map param
 
-/-- `elim_gen_prod n e []` assumes that `e` is an `k`-tuple for `k ≥ n` and inducts on `e` `n`
-  times. Returns the list of newly generated projections, and the resulting tail of `e`. -/
-meta def elim_gen_prod : nat → expr → list expr → tactic (list expr × expr)
-| 0       e hs := return (hs, e)
-| (n + 1) e hs := do
-  [(_, [h, h'], _)] ← induction e [],
-  elim_gen_prod n h' (hs ++ [h])
+meta def mk_exists_lst (args : list expr) (inner : expr) : tactic expr :=
```

Comment:
Wait, I'm confused about where this and `mk_op/and/or_lst` are coming from. These weren't changed in the original diff (and I don't see them being added in any commit). `mk_exists_lst` is in `tactic.core` in master. What mathlib version is this PR based on?

`mk_exists_lst` already exists with a docstring in `tactic.core`, but the others don't (as far as I can tell), so they need doc strings.

Reply:
>What mathlib version is this PR based on?

The parent of the first commit is b11f0f1 which is from Sep. 18. It looks like `mk_*_lst` were moved to `tactic.core` by #1625 (6 days ago).

Reply:
Aha. `mk_exists_lst` moved to `tactic.core` in the `expr` namespace, which is why Travis didn't catch this. The others moved to `meta.expr`. This is very confusing and more evidence that we should be careful about mentioning renamed lemmas in commit messages.

Reply:
I moved them out, since they have nothing to with coinductive predicates. As said in my other comment, much more constructions on `expr` should be moved out of this file (including this one).

Reply:
I 100% agree with the decision to move them -- I've just been a little frustrated recently with the squash-and-merge strategy. It's not clear at all that `feat(lint): add priority test for instances that always apply` renames `tactic.mk_exists_lst` to `expr.mk_exists_list`. I'm definitely guilty of making hidden changes like this myself, and the annoyance of stacked PRs makes it hard to avoid. I don't know what the solution is.

Reply:
Yeah. One idea would be to make the PRs piecemeal but then we end up with millions of PRs, many of which are pretty boring.

Reply:
I like boring PRs. I can review those on the train, and push them to the merge queue. I find PRs that do something substantial, but also make 13 "boring" changes a bit complicated to review. I sometimes wish I could do partial merges (this file is fine, this chunk looks good but other part of the file needs some work, etc...) Small boring PRs make life as a reviewer easy, I think.